addons/crm/crm_lead.py

```diff
@@ -651,147 +628,142 @@ def merge_opportunity(self, cr, uid, ids, user_id=False, team_id=False, context=
+        return highest
 
-        return highest.id
-
-    def _convert_opportunity_data(self, cr, uid, lead, customer, team_id=False, context=None):
-        crm_stage = self.pool.get('crm.stage')
+    @api.one
+    def _convert_opportunity_data(self, customer, team_id=False):
         contact_id = False
         if customer:
-            contact_id = self.pool.get('res.partner').address_get(cr, uid, [customer.id])['default']
+            customer.address_get()['default']
         if not team_id:
-            team_id = lead.team_id and lead.team_id.id or False
+            team_id = self.team_id and self.team_id.id or False
         val = {
-            'planned_revenue': lead.planned_revenue,
-            'probability': lead.probability,
-            'name': lead.name,
+            'planned_revenue': self.planned_revenue,
+            'probability': self.probability,
+            'name': self.name,
             'partner_id': customer and customer.id or False,
             'type': 'opportunity',
-            'date_action': fields.datetime.now(),
-            'date_open': fields.datetime.now(),
-            'email_from': customer and customer.email or lead.email_from,
-            'phone': customer and customer.phone or lead.phone,
-            'date_conversion': fields.datetime.now(),
+            'date_action': fields.Datetime.now(),
+            'date_open': fields.Datetime.now(),
+            'email_from': customer and customer.email or self.email_from,
+            'phone': customer and customer.phone or self.phone,
+            'date_conversion': fields.Datetime.now(),
         }
-        if not lead.stage_id or lead.stage_id.type=='lead':
-            val['stage_id'] = self.stage_find(cr, uid, [lead], team_id, [('type', 'in', ('opportunity', 'both'))], context=context)
+        if not self.stage_id or self.stage_id.type=='lead':
+            val['stage_id'] = self.stage_find(team_id, domain=[('type', 'in', ('opportunity', 'both'))])
         return val
 
-    def convert_opportunity(self, cr, uid, ids, partner_id, user_ids=False, team_id=False, context=None):
+    @api.multi
+    def convert_opportunity(self, partner_id, user_ids=False, team_id=False):
         customer = False
         if partner_id:
-            partner = self.pool.get('res.partner')
-            customer = partner.browse(cr, uid, partner_id, context=context)
-        for lead in self.browse(cr, uid, ids, context=context):
+            customer = self.env['res.partner'].browse(partner_id)
+        for lead in self:
             # TDE: was if lead.state in ('done', 'cancel'):
             if (lead.probability == 100 or lead.probability == 0) and lead.stage_id.on_change and lead.stage_id.sequence > 1:
                 continue
-            vals = self._convert_opportunity_data(cr, uid, lead, customer, team_id, context=context)
-            self.write(cr, uid, [lead.id], vals, context=context)
-
+            vals = lead._convert_opportunity_data(customer, team_id)[0]
```

Comment:
Use @api.multi with self.ensure_one() instead of @api.one to avoid the side effect in return values .